pkg/clients/subnetwork/subnetwork.go

```diff
@@ -42,28 +50,73 @@ func GenerateSubnetwork(s v1alpha3.SubnetworkParameters) *compute.Subnetwork {
 	return sn
 }
 
-// GenerateGCPSubnetworkStatus creates a GCPSubnetworkStatus object using *googlecompute.Subnetwork.
-func GenerateGCPSubnetworkStatus(in *compute.Subnetwork) v1alpha3.GCPSubnetworkStatus {
-	s := v1alpha3.GCPSubnetworkStatus{}
-	s.Name = in.Name
-	s.Description = in.Description
-	s.EnableFlowLogs = in.EnableFlowLogs
-	s.Fingerprint = in.Fingerprint
-	s.IPCIDRRange = in.IpCidrRange
-	s.Network = in.Network
-	s.PrivateIPGoogleAccess = in.PrivateIpGoogleAccess
-	s.Region = in.Region
-	for _, val := range in.SecondaryIpRanges {
-		obj := &v1alpha3.GCPSubnetworkSecondaryRange{
-			IPCidrRange: val.IpCidrRange,
+// GenerateSubnetworkForUpdate creates a *googlecompute.Subnetwork object using
+// SubnetworkParameters with fields disallowed by the GCP API removed. If a
+// field can be included in the GCP API but will result in an error if the value
+// is changed, it will still be included here such that users are notified of
+// invalid updates.
```

Comment:
@negz @muvaf this is the case I was discussing at stand-up.

Reply:
I suspect you made the right call here; this keeps the behaviour as close as possible to the GCP API and is more informative to the user than silently swallowing the update.  Will it mean all updates (of other fields) will fail until the user corrects the problem?

Reply:
@negz yes I believe that would be the case. Not sure how we could get around that though because I don't think we can say "including this field will break the update so we just removed it and sent the rest".

Reply:
Until we have a validation around immutability, I think it's better to return an error saying `GCP does not allow updating this field` because otherwise, they might be tricked into thinking that the update was successful.

For example, in AWS you got `ModifyRequest`s where it only has the update-able fields but this disables Crossplane to say that the field cannot be updated because it never knows that the field changed. It just goes through the fields that have a match in `ModifyRequest` and send that request as opposed to GCP allowing all fields but rejecting the requests depending what fields and values you sent.

Reply:
> they might be tricked into thinking that the update was successful

Not sure I follow here. If the field is immutable the update will fail. We are just deferring to the error that GCP gives us rather than providing our own. It sounds like AWS behaves differently though? Do you just desire returning an error manually here to be more consistent across stacks?